packages/story-editor/src/components/mediaRecording/provider.js

```diff
@@ -143,12 +143,23 @@ function MediaRecordingProvider({ children }) {
     // granted permission, so it's not easy to detect.
     // TODO: Figure out how to retry without microphone if possible.
     mediaStreamConstraints: {
-      audio: audioInput && hasAudio ? { deviceId: audioInput } : true,
-      video: videoInput && hasVideo ? { deviceId: videoInput } : true,
+      audio: hasAudio ? (audioInput ? { deviceId: audioInput } : true) : false,
+      video: hasVideo ? (videoInput ? { deviceId: videoInput } : true) : false,
     },
     onStop,
   });
 
+  useEffect(() => {
+    if (
+      error?.name === 'NotFoundError' ||
+      error?.name === 'NotReadableError' ||
+      error?.name === 'OverConstrainedError'
```

Comment:
IIRC in some browsers this error can occur even when the user has not yet granted permission, so I hope this does not cause any issue in some edge cases. Just noting it here for visibility.

Reply:
I haven't seen it anywhere, for me, it always looks like this:
https://github.com/GoogleForCreators/web-stories-wp/pull/11939#issuecomment-1188305242